packages/core/admin/admin/src/pages/SettingsPage/pages/Roles/ListPage/components/RoleRow/index.js

```diff
@@ -19,9 +19,9 @@ const RoleRow = ({ id, name, description, usersCount, icons, rowIndex }) => {
     <Tr
       aria-rowindex={rowIndex}
       key={id}
-      {...onRowClick({
+      {...(icons.length > 0 && icons[1] ? onRowClick({
         fn: icons[1].onClick,
-      })}
+      }) : {})}
```

Comment:
This feels very "mysterious" is there a way we can make it clearer what we're expecting here? e.g move the check out of the render with the name `canUpdateOrCreate`?

Reply:
ok you are right it is not obvious

Reply:
let me know if you like it, I decided to pass canUpdate as prop because it is the best way to recognize if you can click or not to update a role and to destructure the icons array to take just the part related to the update